packages/@aws-cdk/aws-appmesh/lib/gateway-route-spec.ts

```diff
@@ -143,28 +360,91 @@ class HttpGatewayRouteSpec extends GatewayRouteSpec {
+   * When disabled, retains the original hostname/prefix from the request.
+   *
+   * @default true
+   */
+  readonly rewriteRequestHostname?: boolean;
+
+  /**
+   * The path to rewrite.
+   *
+   * @default - rewritten to the target Virtual Service's hostname and the matched prefix is rewritten to '/'.
+   */
+  readonly pathRewrite?: HttpGatewayRoutePathRewrite;
+
   constructor(options: HttpGatewayRouteSpecOptions, protocol: Protocol.HTTP | Protocol.HTTP2) {
     super();
     this.routeTarget = options.routeTarget;
     this.routeType = protocol;
     this.match = options.match;
+    this.rewriteRequestHostname = options.rewriteRequestHostname;
+    this.pathRewrite = options.pathRewrite;
   }
 
-  public bind(_scope: Construct): GatewayRouteSpecConfig {
-    const prefixPath = this.match ? this.match.prefixPath : '/';
-    if (prefixPath[0] != '/') {
-      throw new Error(`Prefix Path must start with \'/\', got: ${prefixPath}`);
+  public bind(scope: Construct): GatewayRouteSpecConfig {
+    const pathRewriteConfig = this.pathRewrite?.bind(scope);
+    const pathMatchConfig = this.match?.path?.bind(scope).requestMatch;
+
+    // Set prefix Match to '/' if none on match properties are defined.
+    const prefixMatch = areMatchPropertiesUndefined(this.match)
+      ? '/'
+      : pathMatchConfig?.prefix;
+    const prefixPathRewrite = pathRewriteConfig?.prefix;
+    const pathRewrite = pathRewriteConfig?.path;
+
+    if (prefixMatch && prefixMatch[0] != '/') {
+      throw new Error(`Prefix Path for the match must start with \'/\', got: ${prefixMatch}`);
+    }
+    if (prefixPathRewrite?.value && prefixPathRewrite.value[0] != '/') {
+      throw new Error(`Prefix Path for the rewrite must start with \'/\', got: ${prefixPathRewrite.value}`);
     }
+    if (prefixPathRewrite && !prefixMatch) {
+      throw new Error('HTTP Gateway Route Prefix Match must be set.');
+    }
+    if (pathRewrite && prefixMatch) {
+      throw new Error('HTTP Gateway Route Prefix Match and Path Rewrite both cannot be set.');
+    }
+
     const httpConfig: CfnGatewayRoute.HttpGatewayRouteProperty = {
       match: {
-        prefix: prefixPath,
+        prefix: prefixMatch,
+        path: pathMatchConfig?.path,
+        hostname: this.match?.hostname?.bind(scope).hostnameMatch,
+        method: this.match?.method,
+        headers: this.match?.headers?.map(header => header.bind(scope).headerMatch),
+        queryParameters: this.match?.queryParameters?.map(queryParameter => queryParameter.bind(scope).queryParameter),
       },
       action: {
         target: {
           virtualService: {
             virtualServiceName: this.routeTarget.virtualServiceName,
           },
         },
+        rewrite: this.rewriteRequestHostname || pathRewriteConfig
+          ? {
+            hostname: this.rewriteRequestHostname
+              ? {
+                defaultTargetHostname: this.rewriteRequestHostname ? 'ENABLED' : 'DISABLED',
```

Comment:
You check for `this.rewriteRequestHostname` above, so no need to check it again here.

Reply:
Addressed

Reply:
@skinny85 - After rubberducking with teammate, I realized I was only changing HTTP side of code and getting a failure from gRPC test.